_posts/2017-5-31-release-1.3.0.md

```diff
@@ -0,0 +1,104 @@
+
+* **Extended support for aggregations in Table API / SQL**: With Flink 1.3.0, the Table API and SQL support many more types of aggregations, including
+	* GroupBy window aggregations (in SQL via the window functions [TUMBLE, HOP, and SESSION windows](https://issues.apache.org/jira/browse/FLINK-6011)) for both batch and streaming.
+
+	* Over window aggregations (only for streaming).
+
+	* Non-windowed aggregations (in streaming with retractions).
+
+	* User-defined aggregation functions for custom aggregation logic.
+
+→ See [the Flink documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.3/dev/table_api.html#group-windows) for details about these features.
+
+# Connectors
+
+* **ElasticSearch 5.x support**: The ElasticSearch connectors have been restructured to have a common base module and specific modules for ES 1, 2 and 5, similar to how the Kafka connectors are organized. This will make fixes and future improvements available across all ES versions. ([FLINK-4988](https://issues.apache.org/jira/browse/FLINK-4988))
+
+* **Allow rescaling the Kinesis Consumer**: Flink 1.2.0 introduced rescalable state for DataStream programs. With Flink 1.3, the Kinesis Consumer also makes use of that engine feature ([FLINK-4821](https://issues.apache.org/jira/browse/FLINK-4821)).
+
+* **Transparent shard discovery for Kinesis Consumer**: The Kinesis consumer can now discover new shards without failing / restarting jobs when a resharding is happening. ([FLINK-4577](https://issues.apache.org/jira/browse/FLINK-4577))
+
+* **Allow setting custom start positions for the Kafka consumer**: With this change, you can instruct Flink’s Kafka consumer to start reading messages from a specific offset ([FLINK-3123](https://issues.apache.org/jira/browse/FLINK-3123)) or earliest / latest offset ([FLINK-4280](https://issues.apache.org/jira/browse/FLINK-4280)) without respecting committed offsets in Kafka.
+
+* **Allow out-opt from offset committing for the Kafka consumer**: By default, Kafka commits the offsets to the Kafka broker once a checkpoint has been completed. This change allows users to disable this mechanism. ([FLINK-3398](https://issues.apache.org/jira/browse/FLINK-3398))
+
+# CEP Library
+
+The CEP library has been greatly enhanced and is now able to accommodate more use-cases out-of-the-box (expressivity enhancements), make more efficient use of the available resources, adjust to changing runtime conditions--all without breaking backwards compatibility. 
```

Comment:
actually, some things in the API did change!, e.g. `PatternFlatSelectFunction#flatSelect` changed signature